src/EnrollU2FController.js

```diff
@@ -0,0 +1,137 @@
+/*!
+ * Copyright (c) 2015-2016, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+/*global u2f */
+
+define([
+  'okta',
+  'util/FormType',
+  'util/FormController',
+  'views/enroll-factors/Footer',
+  'vendor/lib/q',
+  'u2f-api-polyfill'
+],
+function (Okta, FormType, FormController, Footer, Q) {
+
+  var _ = Okta._;
+
+  return FormController.extend({
+    className: 'enroll-u2f',
+    Model: {
+      local: {
+        '__enrolled__': 'boolean'
+      },
+
+      save: function () {
+        this.trigger('request');
+
+        if (this.get('__enrolled__')) {
+          return this.activate();
+        }
+
+        return this.doTransaction(function (transaction) {
+          var factor = _.findWhere(transaction.factors, {
+            factorType: 'u2f',
+            provider: 'FIDO'
+          });
+          return factor.enroll();
+        });
+      },
+
+      activate: function () {
+        this.set('__enrolled__', true);
+        this.trigger('errors:clear');
+
+        return this.doTransaction(function (transaction) {
+          var activation = transaction.factor.activation;
+          var appId = activation.appId;
+          var request = {
```

Comment:
What is this variable for? Seems like we're not using it (except to construct registerRequests, but at that point why not just do that here?